backend/unit-tests/focs-2-3-tests.ml

```diff
@@ -0,0 +1,148 @@
+(*helper functions*)
+
+let rec intsFromTo x y = 
+if x>y then [] else x::(intsFromTo (x+1) y);;
+
+let intsTo n = 
+let rec intsFrom m =
+if m>n then [] else m::(intsFrom (m+1))
+in
+intsFrom 1 ;;
+
+let removeDupes xs = 
+	let rec aux y xs = match xs with 
+	| [] -> y
+	| x::xs -> aux (x::y) (List.filter (fun i-> i!=x) xs) in
+aux [] xs;; 
+
+let rec rndPerm list =
+    let rec extract acc n = function
+      | [] -> []
+      | x ::xs -> if n = 0 then (x, acc @ xs) else extract (x::acc) (n-1) xs
+    in
+    let extract_rand list len =
+      extract [] (Random.int len) list
+    in
+    let rec aux acc list len =
+      if len = 0 then acc else
+        let picked, rest = extract_rand list len in
+        aux (picked :: acc) rest (len-1)
+    in
+    aux [] list (List.length list);;
+
+(*intersect*)
```

Comment:
Intersection tests don't currently check that the number of cons (::) operations performed does not exceed the number of elements in the result list. Can you override the cons function in the standard library and add a reference to count the number of times it's called?